### ISSUE TITLE
Add staking information to 'your contracts' section

### ISSUE THREAD

**Issue by jvermond** [threefoldtech/tfgrid-sdk-ts#2477]
### Which package/s are you suggesting this feature for?

Dashboard

### Is your feature request related to a problem? Please describe

Currently on top of the contracts list is information on total costs. It would be nice to have a label here related to staking / balance discount. 
![20240319_213756](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/39809174/097d928b-379f-4bb3-9778-0a9bb86a8f9b)
![20240319_213756]

![Screenshot_20240319_213621_Chrome](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/39809174/9983da11-c5f3-4c02-8f9f-8e00150bdbb8)

### Describe the solution you'd like

Some indicator that says:

Current staking discount level: silver (40%)

Or anything similar

**Comment by ramezsaeed** [comment]
Discount is calculated per contract; this means we cannot add it here in general for all contracts as not all contract apply the same discount.

**Comment by ehab-hassan** [comment]
![image](https://github.com/user-attachments/assets/d8c80e52-f667-4716-a134-3b95e60b7487)
Let's show the discount beside the billing rate value and use the same icon we used in Node Finder
@AhmedHanafy725 needs someone to calculate the discount here

**Comment by amiraabouhadid** [comment]
added UI Icon, working on getting discount received logic

**Comment by amiraabouhadid** [comment]
Added graphql retrieval of discount

**Comment by amiraabouhadid** [comment]
Ran manual tests

**Comment by amiraabouhadid** [comment]
found issue when fetching discountReveived with consumption graphql call, debugging

**Comment by amiraabouhadid** [comment]
blocked on #3364

**Comment by amiraabouhadid** [comment]
merged development
running tests by starting out with a default package account, funding account, waiting for package to change after account has been funded

**Comment by amiraabouhadid** [comment]
updated UI and added timestamp_desc in request

**Comment by atefnazmy** [comment]
- Verified on Dev,
- Version: 34cb239
- Verified that the discount icon is now visible and when hovering over it shows the discount whether it is gold or silver or bronze.

![Image](https://github.com/user-attachments/assets/8b3119de-292d-4706-a96e-7a832ff5a599)

![Image](https://github.com/user-attachments/assets/8c3864e6-c0f1-455b-9ec1-12ff8d4fa0e9)

#### Related Link:
- https://app.testlodge.com/a/26076/projects/40893/suites/234374?expand_section=430129#case_4451333

**Comment by ramezsaeed** [comment]
@atefnazmy 
please add two more tests for this https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2477#issuecomment-2564715718
1- from default package up to gold.
2- from gold package down to default.

**Comment by atefnazmy** [comment]
The discount is being applied correctly to the billing rate, but the tooltip displayed an incorrect discount package type when hovering over the discount icon in the contracts table if the contract was deployed before the user's balance stack was updated. Contracts deployed after the balance stack update display the correct discount package type in the tooltip.
 #3873

**Comment by khaledyoussef24** [comment]
can we add also to the tool tip colors of the category instead of the blank label or something?

**Comment by amiraabouhadid** [comment]
> can we add also to the tool tip colors of the category instead of the blank label or something?

can be tracked here https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3902

**Comment by amiraabouhadid** [comment]
please verify after verifying this https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3873

**Comment by atefnazmy** [comment]
- Verified on Dev,
- Version: 85b7dcf
- The stack info has been added to contracts section, and the tooltip displays the correct discount package type

![Image](https://github.com/user-attachments/assets/f86d69c7-093d-4fed-b62d-0f5a4a9992f4)

#### Related Link:
- https://app.testlodge.com/a/26076/projects/40893/suites/234374?expand_section=430129#case_4451333
